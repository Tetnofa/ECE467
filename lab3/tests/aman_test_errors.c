// Test semantic errors (this should fail compilation)
int arr[3];  // Move array declaration to global scope

void errorTest() {
    int x;
    bool y;
    
    // Type mismatch
    x = true;  // Error
    
    // Undefined variable
    z = 5;     // Error
    
    // Invalid operation
    y = x + y; // Error
    
    // Array out of bounds
    arr[5] = 1; // Error
} 