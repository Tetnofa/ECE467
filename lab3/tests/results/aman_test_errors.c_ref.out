Program [useIO=0](2,0)
	Array Declaration	Int[3](2,0)
		Identifier[name:arr](2,4)
	Function[isProto=0]	Void(4,0)
		Identifier[name:errorTest](4,5)
		Scope(4,17)
			Scalar Declaration			Int(5,4)
				Identifier[name:x](5,8)
			Scalar Declaration			Bool(6,4)
				Identifier[name:y](6,9)
			Assignment(9,4)
				Reference(9,4)
					Identifier[name:x](9,4)
				Int Expression(9,8)
					BoolConstant[val=1](9,8)
			Assignment(12,4)
				Reference(12,4)
					Identifier[name:z](12,4)
				Int Expression(12,8)
					IntConstant[val=5](12,8)
			Assignment(15,4)
				Reference(15,4)
					Identifier[name:y](15,4)
				Int Expression(15,8)
					Binary Expression[opcode:+](15,8)
						Int Expression(15,8)
							Reference(15,8)
								Identifier[name:x](15,8)
						Int Expression(15,12)
							Reference(15,12)
								Identifier[name:y](15,12)
			Assignment(18,4)
				Reference(18,4)
					Identifier[name:arr](18,4)
					Int Expression(18,8)
						IntConstant[val=5](18,8)
				Int Expression(18,13)
					IntConstant[val=1](18,13)
sema: 9:4 : type mismatch 
sema: 12:4 : use of undefined identifier z
sema: 12:4 : type mismatch 
sema: 15:8 : type mismatch 
sema: 15:4 : type mismatch 
